PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -982,33 +1082,59 @@ struct LambdaTableProducer {
 
   void processDataRun3(CollisionsRun3::iterator const& collision, aod::V0Datas const& V0s, Tracks const& tracks)
   {
-    fillLambdaTables<kRun3, kData>(collision, V0s, tracks);
+    fillLambdaRecoTables<kRun3, kData>(collision, V0s, tracks);
   }
 
   PROCESS_SWITCH(LambdaTableProducer, processDataRun3, "Process for Run3 DATA", true);
 
   void processDataRun2(CollisionsRun2::iterator const& collision, aod::V0Datas const& V0s, Tracks const& tracks)
   {
-    fillLambdaTables<kRun2, kData>(collision, V0s, tracks);
+    fillLambdaRecoTables<kRun2, kData>(collision, V0s, tracks);
   }
 
   PROCESS_SWITCH(LambdaTableProducer, processDataRun2, "Process for Run2 DATA", false);
 
-  void processMCRecoRun3(soa::Join<CollisionsRun3, aod::McCollisionLabels>::iterator const& collision, aod::McCollisions const&,
-                         McV0Tracks const& V0s, aod::McParticles const&, TracksMC const& tracks)
+  void processMCRecoRun3(soa::Join<CollisionsRun3, aod::McCollisionLabels>::iterator const& collision,
+                         aod::McCollisions const&,
+                         McV0Tracks const& V0s, TracksMC const& tracks,
+                         aod::McParticles const&)
   {
-    fillLambdaTables<kRun3, kMC>(collision, V0s, tracks);
+    fillLambdaRecoTables<kRun3, kMC>(collision, V0s, tracks);
   }
 
   PROCESS_SWITCH(LambdaTableProducer, processMCRecoRun3, "Process for Run3 MC Reconstructed", false);
 
-  void processMCRecoRun2(soa::Join<CollisionsRun2, aod::McCollisionLabels>::iterator const& collision, aod::McCollisions const&,
-                         McV0Tracks const& V0s, aod::McParticles const&, TracksMC const& tracks)
+  void processMCRecoRun2(soa::Join<CollisionsRun2, aod::McCollisionLabels>::iterator const& collision,
+                         aod::McCollisions const&,
+                         McV0Tracks const& V0s, TracksMC const& tracks,
+                         aod::McParticles const&)
   {
-    fillLambdaTables<kRun2, kMC>(collision, V0s, tracks);
+    fillLambdaRecoTables<kRun2, kMC>(collision, V0s, tracks);
   }
 
   PROCESS_SWITCH(LambdaTableProducer, processMCRecoRun2, "Process for Run2 MC Reconstructed", false);
+
+  void processMCGenRun3(aod::McCollisions::iterator const&,
+                        soa::SmallGroups<soa::Join<CollisionsRun3, aod::McCollisionLabels>> const& collisions,
+                        aod::McParticles const& mcParticles)
+  {
+    for (auto const& collision : collisions) {
+      fillLambdaMcGenTables<kRun3>(collision, mcParticles);
+    }
```

Comment:
Be aware that in here your are potentially processing more than one collision reconstructed from a single generated collision and all the reconstructed collisions from the same generated collision go with the same generated particles